src/components/menu/menu-popover.tsx

```diff
@@ -1,14 +1,79 @@
-import { FC, MutableRefObject, useRef } from 'react'
+import { FC, MutableRefObject, useEffect, useRef } from 'react'
 import { useClickOutside } from '../../hooks/use-click-outside'
 import { useMenuContext } from './menu-context'
 import { ElMenuPopover } from './styles'
 
+// close menu when focus move out of menu
+export const menuListFocusOutHandler = (menuList: HTMLDivElement, closeMenu: VoidFunction) => (event) => {
+  if (!menuList.contains((event as FocusEvent).relatedTarget as Node)) {
```

Comment:
**question:** typically the use type assertions is viewed as bad practice, unless there's just no way of getting TS to understand the types involved in our code). Can you help me understand why all these type assertions are needed?

Reply:
apoligize, I forgot to delete it after I removed the event type 🙏